tools/storm/helpers/rebuild_raid.go

```diff
@@ -0,0 +1,491 @@
+	stormtrident "tridenttools/storm/utils/trident"
+
+	"github.com/microsoft/storm"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/crypto/ssh"
+	"gopkg.in/yaml.v2"
+	"libvirt.org/go/libvirtxml"
+)
+
+type RebuildRaidHelper struct {
+	args struct {
+		stormsshconfig.SshCliSettings `embed:""`
+		stormenv.EnvCliSettings       `embed:""`
+		TridentConfigPath             string `help:"Path to the Trident configuration file." type:"string"`
+		DeploymentEnvironment         string `help:"Deployment environment (e.g., bareMetal, virtualMachine)." type:"string" default:"virtualMachine"`
+		VmName                        string `help:"Name of VM." type:"string" default:"virtdeploy-vm-0"`
+		Disk                          string `help:"Disk to fail in RAID array." type:"string" default:"/dev/sdb"`
+		SkipRebuildRaid               bool   `help:"Skip the rebuild RAID step." type:"bool" default:"false"`
+		ArtifactsFolder               string `help:"Folder to copy log files into." type:"string" default:""`
+	}
+
+	failed bool
+}
+
+func (h RebuildRaidHelper) Name() string {
+	return "rebuild-raid"
+}
+
+func (h *RebuildRaidHelper) Args() any {
+	return &h.args
+}
+
+func (h *RebuildRaidHelper) RegisterTestCases(r storm.TestRegistrar) error {
+	r.RegisterTestCase("check-if-needed", h.checkIfNeeded)
+	r.RegisterTestCase("fail-bm-raids", h.failBaremetalRaids)
+	r.RegisterTestCase("replace-vm-raid-disk", h.replaceVirtualMachineRaidDisk)
+	r.RegisterTestCase("check-ssh", h.checkTridentServiceWithSsh)
+	r.RegisterTestCase("rebuild-raid", h.rebuildRaid)
+	return nil
+}
+
+func (h *RebuildRaidHelper) FailFromError(tc storm.TestCase, err error) {
+	h.failed = true
+	tc.FailFromError(err)
+}
+
+// Check if RAID testing is needed based on whether the Trident configuration
+// is set up using RAID storage and not using usr-verity.
+func (h *RebuildRaidHelper) checkIfNeeded(tc storm.TestCase) error {
+	h.failed = false
+
+	tridentConfigContents, err := os.ReadFile(h.args.TridentConfigPath)
+	if err != nil {
+		logrus.Tracef("Failed to read trident config file %s: %v", h.args.TridentConfigPath, err)
+		h.FailFromError(tc, err)
+	}
+	tridentConfig := make(map[string]interface{})
+	err = yaml.UnmarshalStrict(tridentConfigContents, &tridentConfig)
+	if err != nil {
+		logrus.Tracef("Failed to parse trident config file %s: %v", h.args.TridentConfigPath, err)
+		h.FailFromError(tc, err)
+	}
+
+	raidExists := false
+	usrVerity := false
+
+	storage, ok := tridentConfig["storage"].(map[interface{}]interface{})
+	if ok {
+		raidExists = storage["raid"] != nil
+		if verityList, ok := storage["verity"].([]interface{}); ok {
+			if len(verityList) > 0 {
+				usrVerity = verityList[0].(map[interface{}]interface{})["name"] == "usr"
+			}
+		}
+	}
+
+	// TODO (12277): Support for UKI + Rebuild
+	if raidExists && !usrVerity {
+		logrus.Infof("Trident config requires Rebuild testing")
+	} else {
+		logrus.Infof("Trident config does not require Rebuild testing")
+		h.args.SkipRebuildRaid = true
```

Comment:
Note to self: I'll look into implementing a "skip all tests after this" command.